src/js_set.hpp

```diff
@@ -0,0 +1,466 @@
+    NativeAccessor<T> accessor(ctx, *set);
+    return_value.set(set->get(accessor, index));
+}
+
+
+/**
+ * @brief Implements JavaScript Set's add() method
+ * 
+ *  Adds a single element, `A`, of type `T` to the set.  `A` will not be added if it
+ *  already exists within the SetClass.
+ *  See [MDN's reference documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add).
+ * 
+ * @param ctx JS context
+ * @param this_object \ref ObjectType wrapping the SetClass itself
+ * @param args \ref Arguments structure containing a single new element of type `T` to add to the Set
+ * @param return_value \ref ReturnValue wrapping the Set itself, including the newly added element to return to the JS context
+ */
+template<typename T>
+void SetClass<T>::add(ContextType ctx, ObjectType this_object, Arguments &args, ReturnValue &return_value) {
+    args.validate_maximum(1);
+
+    auto set = get_internal<T, SetClass<T>>(ctx, this_object);
+
+    for (size_t i = 0; i < args.count; i++) {
+        validate_value(ctx, *set, args[i]);
+    }
+
+    NativeAccessor<T> accessor(ctx, *set);
+    for (size_t i = 0; i < args.count; i++) {
+        set->insert(accessor, args[i]);
+    }
+
+    return_value.set(this_object);
+}
+
+
+/**
+ * @brief Implements JavaScript Set's `get()` method
+ * 
+ *  Returns a single element found at the given index
+ *  See [MDN's reference documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/get).
+ * 
+ * @param ctx JS context
+ * @param this_object \ref ObjectType wrapping the SetClass itself
+ * @param args \ref Arguments structure containing a single integer
+ * @param return_value \ref ReturnValue wrapping a `Mixed<T>` object, wrapping the found element to return to the JS context
+ */
+template<typename T>
+void SetClass<T>::get(ContextType ctx, ObjectType this_object, Arguments &args, ReturnValue &return_value) {
+    args.validate_maximum(1);
+
+    auto set = get_internal<T, SetClass<T>>(ctx, this_object);
+    // auto size = static_cast<unsigned int>(set->size());
+    // if (size == 0) {
+    //     set->verify_in_transaction();
+    //     return_value.set_undefined();
+    // }
+    // else {
+    //     get_index(ctx, this_object, size - 1, return_value);
+    //     list->remove(size - 1);
+    // }
+
+    // TODO: assert that args[0] is a number
+
+    NativeAccessor<T> accessor(ctx, *set);
+    auto const value_type = set->get_type();
+
+    switch_on_type(value_type, [&](auto const type_indicator) -> void {
+        try {
+            using element_type = std::remove_pointer_t<decltype(type_indicator)>;
+            int requested_index = Value::validated_to_number(ctx, args[0]);
+            realm::Mixed element_value = set->template get<std::remove_pointer_t<decltype(type_indicator)>>(requested_index);
+            return_value.set(element_value);
+        } catch (...) {
+            // TODO:  propagate exception to JS
```

Comment:
If you throw a C++ exception, it will be caught and rewritten to a JS exception. You might not need the `try`/`catch`.

Reply:
You're right.  Fixed the comment.